### PR TITLE
Add verse block

### LINKED ISSUE #1729: Consider renaming "Preformatted" to "Poetry"
Preformatted is a great way to write haikus and poetry, because spaces and linebreaks are respected. 

Should we rename the block to "Poetry", perhaps change the font to be the same as the base font? Or should "poetry" be a separate new block? Or should we add a "style" choice to the block that lets you write with the same font as the body text has, but with the benefits of preformatting? Should we add word-break?

Let's work this out together.

### PR DESCRIPTION
**Pull request by ellatrix**
Open to other names.

Closes #1729.

### CODE REVIEW

File: blocks/library/preformatted/index.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ import Editable from '../../editable';
 const { children } = query;
 
 registerBlockType( 'core/preformatted', {
-	title: __( 'Preformatted' ),
+	title: __( 'Preformatted Text' ),
```
Comment (mtias) [inline blocks/library/preformatted/index.js]: We should fix the display of two lines if we are changing this.

File: blocks/library/verse/index.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+				tagName="pre"
+				value={ content }
+				onChange={ ( nextContent ) => {
+					setAttributes( {
+						content: nextContent,
+					} );
+				} }
+				focus={ focus }
+				onFocus={ setFocus }
+				placeholder={ __( 'Write…' ) }
```
Comment (mtias) [inline blocks/library/verse/index.js]: Should we customize this one a bit: "Add verses or poetry...".

File: blocks/library/verse/style.scss [hunk 3]
```diff
@@ -0,0 +1,8 @@
+pre.wp-block-verse,
+.wp-block-verse pre {
+	white-space: nowrap;
+	font-family: inherit;
+	font-size: inherit;
```
Comment (mtias) [inline blocks/library/verse/style.scss]: Should we change font-size a bit?

Comment (iseulde) [inline blocks/library/verse/style.scss]: What do you have in mind?

Comment (mtias) [inline blocks/library/verse/style.scss]: I am not sure, but some way to differentiate besides margins could be good. Slightly smaller font could give more room for lines to not wrap.

File: blocks/library/verse/index.js [hunk 4]
```diff
@@ -15,7 +15,7 @@ const { children } = query;
 registerBlockType( 'core/verse', {
 	title: __( 'Verse' ),
 
-	icon: 'edit',
+	icon: 'carrot',
```
Comment (iseulde) [inline blocks/library/verse/index.js]: Wait... A carrot? :)

File: blocks/library/verse/index.js [hunk 5]
```diff
@@ -58,6 +58,7 @@ registerBlockType( 'core/verse', {
 				onFocus={ setFocus }
 				placeholder={ __( 'Write…' ) }
 				className={ className }
+				formattingControls={ [ 'bold', 'italic', 'strikethrough' ] }
```
Comment (iseulde) [inline blocks/library/verse/index.js]: Why do we want to exclude link? Only limits possibilities?

File: blocks/library/verse/index.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+import './style.scss';
+import { registerBlockType, createBlock, query } from '../../api';
+import Editable from '../../editable';
+
+const { children } = query;
+
+registerBlockType( 'core/verse', {
+	title: __( 'Verse' ),
+
+	icon: 'carrot',
```
Comment (iseulde) [inline blocks/library/verse/index.js]: Wait... A carrot? :)

File: blocks/library/verse/index.js (hunk 5 above)
Comment (mtias) [inline blocks/library/verse/index.js]: I was talking with @folletto about offering an author/source field instead. I think link detracts from the intention of the block a bit.

File: blocks/library/verse/index.js (hunk 6 above)
Comment (mtias) [inline blocks/library/verse/index.js]: It's a nice icon :) Maybe we can do a custom one (quill?)

File: blocks/library/verse/index.js (hunk 5 above)
Comment (mtias) [inline blocks/library/verse/index.js]: It may also work as an example of how to customize editable. In any case, we can revisit and iterate.

**Comment by mtias** [comment]
Going to merge as it seems a good baseline. We can iterate.

**Comment by ntwb** [comment]
The term `prose` just sprang to mind? 🤔

**Comment by pollyplummer** [comment]
@ntwb Prose is the opposite of verse. :)

**Pull request merged**